### PR TITLE
fix(clustering): upstream use_srv_name cluster compatibility problem

### PR DESCRIPTION
**Pull request by attenuation**
### Summary
https://github.com/Kong/kong/pull/9430 introduce new core entities config `use_srv_name`, but it can not consider compatibility in dp and cp sync config, if DP version is older than CP version. The config will not sync successful.

### Issue reference
FTI-4617

Fix #9430

### CODE REVIEW

**Comment by chronolaw** [comment]
Need a change log entry.

File: kong/clustering/compat/init.lua [hunk 1]
```diff
@@ -324,10 +325,29 @@ function _M.update_compatible_payload(config_table, dp_version, log_suffix)
...
+  if dp_version_num < 3100000000 --[[ 3.1.0.0 ]] then
+    if config_table["upstreams"] then
+      for _, t in ipairs(config_table["upstreams"]) do
+        if t["use_srv_name"] ~= nil then
+          ngx_log(ngx_WARN, _log_prefix, "Kong Gateway v" .. KONG_VERSION ..
+                  " contains configuration 'upstream.use_srv_name'",
+                  " which is incompatible with dataplane version " .. dp_version .. " and will",
+                  " be removed.", log_suffix)
+          t["use_srv_name"] = nil
+          has_update = true
```
Comment (chronolaw) [inline kong/clustering/compat/init.lua:341]: should we break here?

File: kong/clustering/compat/init.lua [hunk 2]
```diff
@@ -324,10 +325,29 @@ function _M.update_compatible_payload(config_table, dp_version, log_suffix)
   if fields then
     config_table = deep_copy(config_table, false)
     if invalidate_keys_from_config(config_table["plugins"], fields, log_suffix) then
-      return true, config_table
+      has_update = true
+    end
+  end
+
+  if dp_version_num < 3100000000 --[[ 3.1.0.0 ]] then
+    if config_table["upstreams"] then
```
Comment (chronolaw) [inline kong/clustering/compat/init.lua:333]: localize `config_upstreams = config_table["upstreams"]`

File: spec/01-unit/19-hybrid/03-compat_spec.lua [hunk 3]
```diff
@@ -302,4 +304,51 @@ describe("kong.clustering.compat", function()
       assert.falsy(check("1.0.0", "1.1.0"))
     end)
   end)
+
+  describe("core entities compatible changes #ttt", function()
```
Comment (chronolaw) [inline spec/01-unit/19-hybrid/03-compat_spec.lua:308]: remove tag `#ttt`

File: kong/clustering/compat/init.lua [hunk 4]
```diff
@@ -324,10 +325,29 @@ function _M.update_compatible_payload(config_table, dp_version, log_suffix)
   if fields then
     config_table = deep_copy(config_table, false)
     if invalidate_keys_from_config(config_table["plugins"], fields, log_suffix) then
-      return true, config_table
+      has_update = true
+    end
+  end
+
+  if dp_version_num < 3100000000 --[[ 3.1.0.0 ]] then
```
Comment (chronolaw) [inline kong/clustering/compat/init.lua:332]: should be `3001000000`?

**Review by chronolaw — APPROVED** [review]

**Review by StarlightIbuki — APPROVED** [review]

**Pull request merged**